### PR TITLE
add knoopx's completionlist to try it out

### PR DESCRIPTION
**Pull request by fdncred**
This is my attempt to add @knoopx's completionlist, by itself, to nana. The theme colors need work. No idea if i did it right but it seems to work. 🤣 

image

I have no idea why the title bar isn't themed, but it didn't work on this windows10 box before this change either.

### CODE REVIEW

**Comment by knoopx** [comment]
looks good but I would suggest to use https://floating-ui.com instead. it is kind a must for rich interfaces and relying on hard-coded values is not good.

File: src/app/Prompt.tsx [hunk 1]
```diff
@@ -14,79 +10,176 @@ type PromptPropType = {
...
+            handleCompletion(e.target.selectionStart);
+          }
+          break;
+      }
+    };
+
+    const hasTopCleareance = () => {
+      if (inputRef.current) {
+        const rect = inputRef.current.getBoundingClientRect();
+        console.log('rect = ', rect);
```
Comment (knoopx) [inline src/app/Prompt.tsx:153]: unnecessary console spamming

Comment (fdncred) [reply src/app/Prompt.tsx:153]: doh! good catch! thanks

**Comment by fdncred** [comment]
> looks good but I would suggest to use https://floating-ui.com instead. it is kind a must for rich interfaces and relying on hard-coded values is not good.

if i knew how to do that, maybe i would. :)

**Comment by knoopx** [comment]
@fdncred hahah fair enough, if you dont feel confident just merge it and ill send a PR whenever I find some time (im in the middle of setting a custom-built keyboard).

if you want to try the examples at https://floating-ui.com/docs/react-dom should get you started (you just need to pass `useFloating` `reference` to `CompletionList` `ref` property and it will automatically calculate overflow based on the actual size). that library also offers some other useful hooks for this specific feature https://floating-ui.com/docs/react-dom-interactions

**Comment by rgwood** [comment]
> I have no idea why the title bar isn't themed, but it didn't work on this windows10 box before this change either.

@fdncred Yeah, sadly [the title bar colour attributes](https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwmwindowattribute) are Windows 11 only. I think it might be possible to do on older Windows versions with undocumented APIs but there be dragons.

**Pull request merged**